src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs

```diff
@@ -275,189 +277,310 @@ internal partial class CodeModel
                 .Concat(Schemas.Groups);
     }
 
-    internal partial class TestDefinitionModel
+    internal partial class OavVariableScope
     {
-
-        [YamlMember(Alias = "useArmTemplate")]
-        public Boolean UseArmTemplate;
+        [YamlMember(Alias = "requiredVariables")]
+        public ICollection<string> RequiredVariables { get; set; } = Array.Empty<string>();
 
         [YamlMember(Alias = "requiredVariablesDefault")]
-        public System.Collections.Generic.Dictionary<string, string> RequiredVariablesDefault;
+        public Dictionary<string, string> RequiredVariablesDefault { get; set; }
+
+        [YamlMember(Alias = "secretVariables")]
+        public ICollection<string>? SecretVariables { get; set; }
```

Comment:
Is this possible to make it non-nullable? Empty collection if we have none makes perfect sense here

Reply:
changed to assign a default empty collection to make it non-nullable